### PR TITLE
Use XDS v3 for SDS

### PR DESCRIPTION
**Pull request by howardjohn**
For https://github.com/istio/istio/issues/19885

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

### CODE REVIEW

File: pkg/bootstrap/option/convert.go [hunk 1]
```diff
@@ -94,6 +95,9 @@ func tlsContextConvert(tls *networkingAPI.ClientTLSSettings, sniName string, met
 		CommonTlsContext: &auth.CommonTlsContext{},
 	}
 
+	// We always set v3, since we know this is a new proxy that supports v3
+	requestedType := v3.ListenerType
```
Comment (ramaraochavali) [inline pkg/bootstrap/option/convert.go:99]: nit: v3.ClusterType may be more appropriate as this is applied for cluster, but does not really matter

File: pkg/bootstrap/option/instances_test.go [hunk 2]
```diff
@@ -719,6 +720,9 @@ func TestOptions(t *testing.T) {
 					g.Expect(ok).To(BeFalse())
 				} else {
 					g.Expect(ok).To(BeTrue())
+					if c.key == "envoy_metrics_service_tls" && actual != c.expected {
```
Comment (ramaraochavali) [inline pkg/bootstrap/option/instances_test.go:723]: is it for debugging?

Comment (howardjohn) [reply pkg/bootstrap/option/instances_test.go:723]: oops, thanks

File: pilot/pkg/security/model/authentication.go [hunk 3]
```diff
@@ -102,15 +103,21 @@ func ConstructSdsSecretConfigWithCustomUds(name, sdsUdsPath string) *tls.SdsSecr
 			InitialFetchTimeout: features.InitialFetchTimeout,
 		},
 	}
+
+	if requestedType == v3.ClusterType || requestedType == v3.ListenerType {
+		// For v3 clusters/listeners, send v3 secrets
+		cfg.SdsConfig.ResourceApiVersion = core.ApiVersion_V3
```
Comment (hzxuzhonghu) [inline pilot/pkg/security/model/authentication.go:109]: Nit: This is the bad part, we can not make sure the requestedType is always the above two as we elvove.

Comment (howardjohn) [reply pilot/pkg/security/model/authentication.go:109]: Is your concern new types? Or new versions (like v4)?

Comment (hzxuzhonghu) [reply pilot/pkg/security/model/authentication.go:109]: both

Comment (howardjohn) [reply pilot/pkg/security/model/authentication.go:109]: XDS v4 will not be for over a year, at least. I don't expect any new resources that refer to secrets in the mean time. Note that this code is purely temporary for 1.7. In 1.8, we expect all clients to be XDS v3 and can remove the compatibility code here

**Pull request merged**